Brokerages/Kraken/KrakenBrokerage.cs

```diff
@@ -0,0 +1,281 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Packets;
+using QuantConnect.Securities;
+using HistoryRequest = QuantConnect.Data.HistoryRequest;
+using Order = QuantConnect.Orders.Order;
+
+namespace QuantConnect.Brokerages.Kraken
+{
+    /// <summary>
+    /// Kraken Brokerage implementation
+    /// </summary>
+    public class KrakenBrokerage : Brokerage, IDataQueueHandler
```

Comment:
Merged the KrakenBrokerage and KrakenApi into KrakenBrokerage because of very same reason :).

Reply:
Awesome! This makes the `KrakenRestApi` the obvious choice for a test seam (hopefully it implements an interface, if not, we can make it implement one and have that be the only edit to that file and use the interface as a mock target)

Reply:
What kind of interface? 

Because I was testing mocks and seems like you can't mock private/internal vars/funcs, isn't it so; is this reason why to implement the interface?